easybuild/tools/modules.py

```diff
@@ -131,6 +133,141 @@
 _log = fancylogger.getLogger('modules', fname=False)
 
 
+class ModuleEnvironmentVariable:
+    """
+    Environment variable data structure for modules
+    Contents of environment variable is a list of unique strings
+    """
+
+    def __init__(self, contents, top_level_file=False, delim=os.pathsep):
+        """
+        Initialize new environment variable
+        Actual contents of the environment variable are held in self.contents
+        """
+        self.contents = contents
+        self.top_level_file = bool(top_level_file)
```

Comment:
That might be misleading. The current behaviour (v4.9) is that all paths require files in them to be declared in the module file. This attribute only changes the strictness of that check to require those files to be in the top directory.

Here there is an underlying design question, because `ModuleEnvironmentVariable` can be generic in purpose and used for other things than `make_module_req`. So maybe we need a ternary parameter (NO_FILES, ANY_FILES, TOP_LEVEL_FILES).

Reply:
Got a nice solution for this, I generalized a bit more `ModuleEnvironmentVariable` by adding a _type_ attribute to it which can be one of `ModEnvVarType`. A new enum with `STRING`, `PATH`, `PATH_WITH_FILES` and `PATH_WITH_TOP_FILES`. Then with a single attribute we can control the behaviour of a specific variable. See https://github.com/easybuilders/easybuild-framework/pull/4653/commits/516d6e659df0482ea4ac75a4a8a379d71b2ebf60

Reply:
Improved further the use and tests of `ModEnvVarType` in https://github.com/easybuilders/easybuild-framework/pull/4653/commits/d4a9df04bac44d7cb5067a8ffa7395f89ce54481

This is ready on my side.